src/library/util.js

```diff
@@ -134,6 +144,31 @@ export const sleep = async s => new Promise(r => setTimeout(() => { r(); }, s));
  * @returns {Boolean}
  */
 // eslint-disable-next-line max-len
-export const itemExists = (obj, param) => typeof obj === 'object' && obj !== null ? Object.prototype.hasOwnProperty.call(
-  obj, param,
-) : false;
+export const itemExists = (obj, param) =>
+  typeof obj === 'object' && obj !== null ? Object.prototype.hasOwnProperty.call(obj, param) : false;
+
+/**
+ * Checks to see if there is a sharedMicroApplication data store available for the Account or User objects
+ * @param {*} sharedMicroApplicationData the object as is, from the Account or User entities
+ * -- value each individual microApplication data store
+ * @param {*} service the service name (serverless name) of the current processing service
+ * @returns the destructured sharedMicroApplicationData
+ * @abstract the "microApplicationsToShareWith" array is a required in order to share this data set with other Micro Applications.
+ * If this field is omitted this function assumes the value is meant to be private.
+ * If the array does exist and there is a "*" in the array, then the value should be shared with all other micro applications
+ */
+export const evaluateSharedMicroApplicationData = (sharedMicroApplicationData, service) => {
+  const readableSharedMicroApplicationData = {};
+
+  // eslint-disable-next-line
+  for (const [key, value] of Object.entries(sharedMicroApplicationData)) {
+    if (
+      (value.microApplicationsToShareWith.length && value.microApplicationsToShareWith.includes(service)) ||
+      value.microApplicationsToShareWith.includes('*')
+    ) {
+      readableSharedMicroApplicationData[key] = service === key ? value : value.serviceData;
```

Comment:
trying to reverse engineer the structure of sharedMicroApplicationData object from this line
```
   {
     "antivirus": {
       "microApplicationsToShareWith": ["*"],
       "serviceData": { "hello": true }
     },
     "gmb": {
        "microApplicationsToShareWith": ["antivirus"]
        "serviceData": { "locationId": "xxx" }
      }
   }
```

Then when antivirus service evaluates this it ends up with 
```
{
     "antivirus": {
       "microApplicationsToShareWith": ["*"],
       "serviceData": { "hello": true }
     },
     "gmb": {
        "locationId": "xxx"
      }
   }
```

Is that right?  If so - it feels a little inconsistent to me that the locationId field gets unnested from serviceData.  I get that we would want to strip the "microApplicationsToShareWith" key since it wouldn't be writable but the non-originating service, but I think I would lean on always keeping `readableSharedMicroApplicationData[key].serviceData` as the path.   Oh - and should this key really be called "serviceData" or should we make an early switch to "appData" or something?

Reply:
So in this case "value" is the entire "micro application" data store: 

```{
  "microApplicationsToShareWith": [],
  "serviceData": {}
}

Reply:
I'm conflicted as to whether or not we call it "serviceData" or "appData" but I also had that thought. The reason I landed on "serviceData" is because "Micro Application" can entail the entire product definition, from front-end component to the micro service itself. But it's a good callout and I think we should consult Ross, Ryan and Bharath on this

Reply:
Are you sure about that? what is this ternary here doing then `readableSharedMicroApplicationData[key] = service === key ? value : value.serviceData;
`

Reply:
This ternary is checking to see if current processing service is the owner of the item in the iteration. If it is, it returns both the `serviceData`, and `readableSharedMicroApplicationData` properties so that the service workers can overwrite the `readableSharedMicroApplicationData`. If it's not the owner, then it only returns the serviceData to read. Although... this made me realize that I need to add some code to the "writing" function in process.js to ensure that only the owner can overwrite the serviceData too.